docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,562 @@
+# Diamond workflow support
+
+A diamond workflow allows for a pipeline to scale out the generation of a single bundle.
+
+A diamond is composed of multiple splits; each split tries to write a set of files.
+Ideally, 2 splits do not write the same file.
+
+To elaborate, a bundle consists of a filesystem namespace with files in folders; each split can write files to any
+folder but should not try to write the same file.
+
+The individual nodes add files to splits without any co-ordination between them.
+
+Commit logic handles any conflicts that exist. Conflicts can only occur if the same file in the namespace of
+the bundle is written to by more than one split with different data.
+
+Diamonds and splits are essentially transitory objects, which live in the metadata namespace of a datamon repo,
```

Comment:
according to `defaultDiamond` in #403 , diamonds exist in `context.Stores.VMetadata`, the version metadata, rather than `context.Stores.Metadata`.

i probably am finicky about homonyms and that sort of thing lately, yet also it probably is important to be very clear about whether we're talking about `VMetadata` or `Metadata` in conversation since these two names are nearly the same and mean different things (i am still not totally clear on `VMetadata`'s meaning as distinct from `Metadata` other than the summary here about it being safe to delete on a GC-like basis).

Reply:
The way I seed it is that versioned metadata is a specialization of metadata.